### PR TITLE
These changes add support for checkstyle in Eclipse.

### PR DESCRIPTION
**Pull request by childsb**
* Removed eclipse formatter settings
* Added configuration for Eclipse Checkstyle plugin
* Generate Eclipse code format rules based on checkstyle settings

Build should work as before.  Install the Eclipse Checkstyle plugin to take advantage of the configuration and formatter.

### CODE REVIEW

**Comment by haoyuan** [comment]
@childsb  Are you still working on this PR? Thanks.

**Comment by childsb** [comment]
@haoyuan when I left this bug it was good but i haven't tested it in a few months.  i wont be able to circle back around to it for another week or so.

**Comment by haoyuan** [comment]
@childsb There was a comment never fixed which blocked the PR. It will be great to move this.

**Comment by childsb** [comment]
i removed the whitespace nit hsaputra mentioned... didn't realize that was blocking this bug @haoyuan

File: core/pom.xml [hunk 1]
```diff
@@ -427,6 +424,30 @@
           </dependency>
         </dependencies>
       </plugin>
+      <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (haoyuan) [inline core/pom.xml]: indentation is not right.

**Comment by jsimsa** [comment]
Jenkins, test this please.

**Comment by jsimsa** [comment]
Can you please fix the merge conflicts?

**Comment by calvinjia** [comment]
@childsb 
Could you resubmit this with the latest master? I'm going to close this PR as it is out of date, but it would be good to move to a check style that is not eclipse specific.

**Pull request closed without merging**